extensions/interactions/TextInput/webdriverio.js

```diff
@@ -0,0 +1,98 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview End-to-end testing utilities for the Text interaction.
+ */
+
+var waitFor = require(
+  process.cwd() + '/core/tests/webdriverio_utils/waitFor.js');
+var action = require(
+  process.cwd() + '/core/tests/webdriverio_utils/action.js');
+var objects = require(process.cwd() + '/extensions/objects/webdriverio.js');
+
+var customizeInteraction = async function(elem, placeholderText, heightOfBox) {
+  await objects.UnicodeStringEditor(
+    await elem.$('<schema-based-unicode-editor>')
+  ).setValue(placeholderText);
+  await objects.IntEditor(
+    await elem.$('<schema-based-int-editor>')
+  ).setValue(heightOfBox);
+};
+
+var expectInteractionDetailsToMatch = async function(
+    elem, placeholderText, heightOfBox) {
+  const textInputInteraction = $(
+    '<oppia-interactive-text-input>');
+  // We use presenceOf here instead of visibilityOf because the container
+  // has a height and width of 0.
+  await waitFor.presenceOf(
+    textInputInteraction,
+    'TextInput interaction taking too long to appear');
+  if (placeholderText) {
+    placeholderValue = await textInputInteraction.getAttribute(
+      'placeholder-with-value');
+    placeholderValueUnicode = JSON.parse(
+      placeholderValue.replace(/&quot;/g, '"')
+    ).unicode_str;
+    expect(placeholderValueUnicode).toEqual(placeholderText);
+  }
+  if (heightOfBox) {
+    expect(textInputInteraction.getAttribute('rows-with-value'))
+      .toEqual(heightOfBox.toString());
+  }
+  expect(
+    await elem.$('<oppia-interactive-text-input>').isExisting()
+  ).toBe(true);
+};
+
+var submitAnswer = async function(elem, answer) {
+  // Try to get the text area element. If it doesn't exist, try input instead.
+  // They are different depending on the height of the box.
+  var textInputElem = elem.$('<oppia-interactive-text-input>');
+  var textAreaElem = textInputElem.$('<textarea>');
+  var inputElem = textInputElem.$('<input>');
+  if (await textAreaElem.isExisting()) {
+    await textAreaElem.setValue(answer);
+    var submitAnswerBtn = $('.e2e-test-submit-answer-button');
+    await action.click('Submit Answer Button', submitAnswerBtn);
+  } else {
+    // This must be chained in here due to the textInputElem possibly being
+    // invisible after the longer text area submits, causing the instantiation
+    // of this promise object to throw a validation error due to it referring
+    // to an element which does not exist.
+    if (await inputElem.isExisting()) {
+      await inputElem.clearValue();
+      await inputElem.setValue(answer);
```

Comment:
In protractor also we are directly using the sendKeys on element instead of using action.js file, so I also did the same.
We cannot use keys with elements in webdriverio, basically we use keys with browser and it type the input in the active element at that point, hence keys is not a subsitute for sendKeys instead setValue as told by you is the actual substitue.

Reply:
Using action.__ is better way because it waits for the visibility of the element too, right? Hence making the tests less flaky.

Reply:
Done.